resources/assets/js/components/FindSkillsModal.tsx

```diff
@@ -101,6 +120,34 @@ const FindSkillsModal: React.FunctionComponent<FindSkillsModalProps> = ({
   // Used to set the button color of an active skill category.
   const [buttonClicked, setButtonClicked] = useState("");
 
+  /**
+   * Filters through the skills list for the any skills that match the search query.
+   * @param searchQuery The search query entered by the user.
+   */
+  const handleSkillSearch = (searchQuery: string): Promise<void> => {
+    const skillNames: string[] = skills.map((skill) =>
+      localizeFieldNonNull(locale, skill, "name"),
+    );
+    const skillStrings: string[] = matchStringsCaseDiacriticInsensitive(
+      searchQuery,
+      skillNames,
+    );
+    const skillMatches = skills.filter((skill) =>
+      skillStrings.includes(localizeFieldNonNull(locale, skill, "name")),
+    );
+
+    // Set the skillResults state with the matches from the query.
+    setFirstVisit(false);
+    setResultsSectionText({
+      title: intl.formatMessage(messages.searchResultsTitle, {
+        numOfSkills: skillMatches.length,
+        searchQuery,
+      }),
+      description: "",
+    });
+    setSkillsResults(skillMatches);
+    return Promise.resolve();
+  };
   return (
     <section>
       <Dialog.Trigger
```

Comment:
What the trigger looks like seems like it might be something the parent of FindSkillsModal might want to control. The fact that there's a placeholder image here right now kind of proves that this component doesn't know what the trigger should look like. Lets replace it with a `triggerContent: React.ReactElement` prop.

Reply:
Looking closer at this, I see that the properties on the Dialog.Trigger component itself are pretty important. So to allow the parent to control what the trigger looks like, we'd either need two props, like so:
```
 <Dialog.Trigger id="findSkills" {...triggerProps}>
    {triggerContent}
 </Dialog.Trigger>
 ```
 Or allow the parent to pass in the whole thing, almost like a `{children}` slot. That would suggest treating FindSkillsModal as _just_ the Dialog component, passing in the id from the parent, and probably allowing the parent to own the Dialog.Overlay as well:
 ```
  <Dialog.Trigger id="findSkills">
    ...
 </Dialog.Trigger>
 <FindSkillsModal id="findSkills" ... />
 <Dialog.Overlay />
 ```

Reply:
Yup, that's good idea 👍 I think the latter of the two options is better. Have a prop called something like `triggerContent` and pass the `string | React.ReactElement` in there. However, I think the overlay should stay with the Dialog component, I don't see any advantage having the parent components having to remember to add that piece in. In fact, we could probably just get rid of the overlay sub-component completely and just add it to the main Dialog component, so we don't have to worry about it.

Reply:
That's a good point! I'm all in favour of adding the overlay to the main Dialog component, as long as we can confirm those no issue with having multiple Overlays in a page.